daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env sh
+
+# Valid actions:
+
+# - FATAL       - netdata exited due to a fatal condition
+#      ACTION_RESULT  -- program name and thread tag
+#      ACTION_DATA    -- fmt, args passed to fatal
+# - START       - netdata started
+#      ACTION_DATA     -- nan
+# - EXIT     - installation action
+#      ACTION_DATA    -- ret value of
+
+ACTION="${1}"
+ACTION_RESULT="${2}"
+ACTION_DATA="${3}"
+
+# -------------------------------------------------------------------------------------------------
+# check opt-out
+
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "/etc/netdata" "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            exit 0
+        fi
+    done
+else
+    if [ -f "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics" ]; then
+        exit 0
+    fi
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION="unknown"
+NAME="unknown"
+VERSION="unknown"
+VERSION_ID="unknown"
+ID="unknown"
+ID_LIKE="unknown"
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+fi
+
+if [ "${NAME}" == "unknown" -o "${VERSION}" == "unknown" -o "${ID}" == "unknown" ] ; then
```

Comment:
koalaman/shellcheck/wiki/SC2166 is still valid.

```suggestion
if [ "${NAME}" == "unknown" ] || [ "${VERSION}" == "unknown" ] || [ "${ID}" == "unknown" ] ; then
```